arcgis-ios-sdk-samples/Features/Display feature layers/DisplayFeatureLayersViewController.swift

```diff
@@ -0,0 +1,189 @@
+        let featureLayer = AGSFeatureLayer(item: item, layerID: 0)
+        // Set the viewpoint to Portland, Oregon.
+        let viewpoint = AGSViewpoint(latitude: 45.5266, longitude: -122.6219, scale: 6000)
+        setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+    }
+    
+    /// Load a feature layer with a local geodatabase.
+    func loadGeodatabase() {
+        // Instantiate the geodatabase with the file name.
+        geodatabase = AGSGeodatabase(name: "LA_Trails")
+        
+        // Load the geodatabase for feature tables.
+        geodatabase.load { [weak self] (error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Get the feature table with the name.
+                let featureTable = self.geodatabase.geodatabaseFeatureTable(withName: "Trailheads")!
+                // Create a feature layer with the feature table.
+                let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                // Set the viewpoint to Malibu, California.
+                let viewpoint = AGSViewpoint(center: AGSPoint(x: -13214155, y: 4040194, spatialReference: .webMercator()), scale: 35e4)
+                self.setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+            }
+        }
+    }
+    
+    /// Load a feature layer with a local geopackage.
+    func loadGeopackage() {
+        // Create a geopackage from a named bundle resource.
+        geoPackage = AGSGeoPackage(name: "AuroraCO")
+        
+        // Load the geopackage.
+        geoPackage?.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            }
+            // Add the first feature layer from the geopackage to the map.
+            if let featureTable = self.geoPackage?.geoPackageFeatureTables.first {
+                // Create the feature layer with the feature table.
+                let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                // Set the viewpoint to Aurora, Colorado.
+                let viewpoint = AGSViewpoint(latitude: 39.7294, longitude: -104.8319, scale: 577790.554289)
+                self.setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+            }
+        }
+    }
+    
+    /// Load a feature layer with a local shapefile.
+    func loadShapefile() {
+        // Create a shapefile feature table from a named bundle resource.
+        let shapefileTable = AGSShapefileFeatureTable(name: "ScottishWildlifeTrust_ReserveBoundaries_20201102")
+        
+        // Create a feature layer for the shapefile feature table.
+        let featureLayer = AGSFeatureLayer(featureTable: shapefileTable)
+        // Ensure the feature layer's metadata is loaded.
+        featureLayer.load { [weak self] error in
```

Comment:
I realize that the portal item and URL examples above show the implicit layer loading mechanism, whereas here you put the set viewpoint method in the loaded block. Should we prefer one pattern over the other? Just a thought.